pkg/catalogsourceconfig/catalogsourcebuilder.go

```diff
@@ -28,13 +28,29 @@ func (b *CatalogSourceBuilder) WithTypeMeta() *CatalogSourceBuilder {
 // WithMeta sets basic TypeMeta and ObjectMeta.
 func (b *CatalogSourceBuilder) WithMeta(name, namespace string) *CatalogSourceBuilder {
 	b.WithTypeMeta()
-	b.cs.ObjectMeta = metav1.ObjectMeta{
-		Name:      name,
-		Namespace: namespace,
+	objectMeta := b.cs.GetObjectMeta()
+	if objectMeta == nil {
+		b.cs.ObjectMeta = metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		}
+	} else {
+		// This is to account for the scenario where WithVisibilityMeta() was called before
+		b.cs.SetName(name)
+		b.cs.SetNamespace(namespace)
 	}
 	return b
 }
 
+// WithVisibilityMeta sets basic TypeMeta and ObjectMeta with the "olm-visibility" label.
+func (b *CatalogSourceBuilder) WithVisibilityMeta(name, namespace string) *CatalogSourceBuilder {
+	b.WithTypeMeta()
+	b.WithMeta(name, namespace)
```

Comment:
Is it preferable to define this function call here like this? Wouldn't it be better to do something similar to what's done in the `WithMeta()` method? i.e.

`if objectMeta == nil {
  b.cs.ObjectMeta{
    Labels: map[string]string{"olm-visibility": "hidden"}
  }
} else {
  b.cs.ObjectMeta.Labels = map[string]string{"olm-visibility": "hidden"}
}`

I know that the reason you are doing this here is that you are right now guaranteeing that some fields will be set, but if those methods are ever changed then it may be better to only modify the fields that the function is described to do.

Reply:
So you want me to remove the `b.WithMeta()` call and only define this function as `WithVisibilityLabel()`? I am fine doing that if that is what you are asking. @tkashem are you in agreement too?

Reply:
Yeah, that is what I mean. I think it makes sense that the function only does one thing.

Reply:
I think that's a good idea to add `WithVisibilityLabel()` that adds the label.